src/Raven.Client/Documents/BulkInsert/BulkInsertOperation.cs

```diff
@@ -750,35 +748,36 @@ private async Task AppendAsyncInternal(DateTime timestamp, IEnumerable<double> v
 
                         if (_first == false)
                         {
-                            _operation._currentWriter.Write(",");
+                            _operation.WriteComma();
                         }
 
                         _first = false;
 
-                        _operation._currentWriter.Write("{\"Timestamp\":\"");
-                        _operation._currentWriter.Write(timestamp.GetDefaultRavenFormat());
-                        _operation._currentWriter.Write("\",\"Values\":[");
+                        _operation._currentWriter.Write('[');
+                        _operation._currentWriter.Write(timestamp.Ticks);
+                        _operation.WriteComma();
+
+                        _operation._currentWriter.Write(values.Count);
+                        _operation.WriteComma();
 
                         var firstValue = true;
                         foreach (var value in values)
                         {
                             if (firstValue == false)
-                                _operation._currentWriter.Write(",");
+                                _operation.WriteComma();
 
                             firstValue = false;
-                            _operation._currentWriter.Write(value);
+                            _operation._currentWriter.Write(Convert.ToString(value, CultureInfo.InvariantCulture));
```

Comment:
Do we need to do https://github.com/ravendb/ravendb/blob/7750eec880a0f5b185d11fce76ff74e1d3530fa5/src/Sparrow/Json/BlittableWriter.cs#L101 ?

Reply:
I think that `EnsureDecimalPlace` is wrong?
take a look at this: https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#GFormatString

Reply:
We switched from G17 to R, because R is a round-trip format

Reply:
Not according to the docs, https://docs.microsoft.com/en-us/dotnet/standard/base-types/standard-numeric-format-strings#RFormatString:
> For Double values, the "R" format specifier in some cases fails to successfully round-trip the original value

Reply:
Please open an issue and assign it to yourself. In the meanwhile, I think that we need to use exactly the same technique there.

Reply:
done

Reply:
okay.. https://issues.hibernatingrhinos.com/issue/RavenDB-14902